rmgpy/data/thermo.py

```diff
@@ -1404,6 +1422,38 @@ def getThermoDataFromGroups(self, species):
         findCp0andCpInf(species, thermoData)
         return thermoData
 
+    def get_thermo_data_from_ml(self, species, ml_estimator, ml_uncertainty_cutoffs):
+        """
+        Return the set of thermodynamic parameters corresponding to a
+        given :class:`Species` object `species` by estimation using the
+        ML estimator. Also compare the estimated uncertainties to the
+        user-defined cutoffs. If any of the uncertainties are larger
+        than their corresponding cutoffs, return None.
+
+        The entropy is not corrected for the symmetry of the molecule.
+        This should be done later by the calling function.
+        """
+        if species.molecule[0].isRadical():
+            thermo = [self.estimateRadicalThermoViaHBI(mol, ml_estimator.get_thermo_data) for mol in species.molecule]
+            thermo0 = min(thermo, key=lambda tdata: tdata.H298)
```

Comment:
See fixup commit. I only sort it in the case of radicals, because I would like to avoid making additional predictions for all resonance isomers if not necessary. Is it also important to sort the non-radical ones?

Reply:
I think that's ok. All of the resonance structures would end up with the same thermo anyway, right?

The changes I'm making to resonance structure generation in #1498 should make this sorting less important as well.

Reply:
Yup, that's right.